mlx/fft.cpp

```diff
@@ -188,4 +191,122 @@ array irfftn(const array& a, StreamOrDevice s /* = {} */) {
   return fft_impl(a, true, true, s);
 }
 
-} // namespace mlx::core::fft
+array stft(
```

Comment:
Don't these need to be defined in the header? Does the code currently compile?

I highly recommend building and running the tests locally. It will shorten the number of times we have to go back and forth on review.